src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusMetric.cs

```diff
@@ -0,0 +1,266 @@
+// <copyright file="PrometheusMetric.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Text;
+
+namespace OpenTelemetry.Exporter.Prometheus;
+
+internal sealed class PrometheusMetric
+{
+    public PrometheusMetric(string name, string unit, PrometheusType type)
+    {
+        // The metric name is
+        // required to match the regex: `[a-zA-Z_:]([a-zA-Z0-9_:])*`. Invalid characters
+        // in the metric name MUST be replaced with the `_` character. Multiple
+        // consecutive `_` characters MUST be replaced with a single `_` character.
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/b2f923fb1650dde1f061507908b834035506a796/specification/compatibility/prometheus_and_openmetrics.md#L230-L233
+        var sanitizedName = SanitizeMetricName(name);
+
+        string sanitizedUnit = null;
+        if (!string.IsNullOrEmpty(unit))
+        {
+            sanitizedUnit = GetUnit(unit);
+
+            // The resulting unit SHOULD be added to the metric as
+            // [OpenMetrics UNIT metadata](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#metricfamily)
+            // and as a suffix to the metric name unless the metric name already contains the
+            // unit, or the unit MUST be omitted. The unit suffix comes before any
+            // type-specific suffixes.
+            // https://github.com/open-telemetry/opentelemetry-specification/blob/b2f923fb1650dde1f061507908b834035506a796/specification/compatibility/prometheus_and_openmetrics.md#L242-L246
+            if (!sanitizedName.Contains(sanitizedUnit))
+            {
+                sanitizedName = sanitizedName + "_" + sanitizedUnit;
+            }
+        }
+
+        // If the metric name for monotonic Sum metric points does not end in a suffix of `_total` a suffix of `_total` MUST be added by default, otherwise the name MUST remain unchanged.
+        // Exporters SHOULD provide a configuration option to disable the addition of `_total` suffixes.
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/b2f923fb1650dde1f061507908b834035506a796/specification/compatibility/prometheus_and_openmetrics.md#L286
+        if (type == PrometheusType.Counter && !sanitizedName.Contains("total"))
```

Comment:
If the requirement is to check if the string "ends with" `_total`, then probably use `string.EndsWith`?

```suggestion
        if (type == PrometheusType.Counter && !sanitizedName.EndsWith("_total"))
```